srml/support/src/additional_traits.rs

```diff
@@ -46,3 +46,23 @@ impl<T, U> ChargeFee<T> for DummyChargeFee<T, U> {
 	fn charge_fee(_: &T, _: Self::Amount) -> Result<(), &'static str> { Ok(()) }
 	fn refund_fee(_: &T, _: Self::Amount) -> Result<(), &'static str> { Ok(()) }
 }
+
+/// A type which can verify doughnuts for an inherent permission domain
+pub trait DoughnutVerifier<Doughnut> {
+	/// The doughnut permission domain for verification
+	const DOMAIN: &'static str;
+
+	fn verify_doughnut(
+		doughnut: &Doughnut,
+		module: &str,
+		method: &str,
+	) -> Result<(), &'static str>;
```

Comment:
I really don't like all the `&'static str` error string in Substrate.
Can we do something like 
```
enum DoughnutError {
  CustomError(&'static str),
  NotSupported,
  SomeOtherCommonError,
}
```

Reply:
Good idea, I might make error an associated type. Since we want runtimes to implement this bit ad-hoc

Reply:
I won't do this in this PR because it's going to be more complex than I thought since module calls still use `Result<(), &'static str>`